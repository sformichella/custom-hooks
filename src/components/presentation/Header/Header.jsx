import React from 'react';
import PropTypes from 'prop-types';

export default function Header({ title }) {
  return (
    <div>
      <h1>{title}</h1>
      <a href="/">Home</a>
    </div>
  );

}

Header.propTypes = {
  title: PropTypes.string.isRequired
};
