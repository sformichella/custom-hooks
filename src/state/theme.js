import { useState } from 'react';

const useTheme = () => {
  const [theme, setTheme] = useState(false);

  const toggleTheme = () => {
    setTheme(theme => !theme);
  };

  return { theme, toggleTheme };
};

export { useTheme };
