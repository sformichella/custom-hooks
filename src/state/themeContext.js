import React, { createContext, useContext, useEffect, useState } from 'react';

export const ThemeContext = createContext(null);

export const ThemeProvider = ({ children }) => {

};

export const useTheme = () => {
  const { theme } = useContext(ThemeContext);
  return theme;
};
