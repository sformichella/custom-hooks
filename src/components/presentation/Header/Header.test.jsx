import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Header from './Header';
import { ThemeProvider } from '../../../state/themeContext';

describe('Header component', () => {
  afterEach(() => cleanup());
  it('renders Header', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Header title="Home" />
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
