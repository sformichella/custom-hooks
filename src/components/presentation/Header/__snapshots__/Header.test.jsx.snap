// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Home
    </h1>
    <a
      href="/"
    >
      Home
    </a>
  </div>
</DocumentFragment>
`;
